core/cog-platform.c

```diff
@@ -179,28 +173,26 @@ cog_platform_create_im_context(CogPlatform *platform)
  * Since: 0.18
  */
 CogPlatform *
-cog_platform_configure(const char *name, const char *params, const char *env_prefix, CogShell *shell, GError **error)
+cog_platform_configure(const char *params, const char *env_prefix, CogShell *shell, GError **error)
 {
-    g_return_val_if_fail(!default_platform, default_platform);
+    CogPlatform *platform = cog_platform_get_default();
+    if (!platform) {
+        g_set_error_literal(error,
+                            COG_PLATFORM_WPE_ERROR,
+                            COG_PLATFORM_WPE_ERROR_INIT,
+                            "Failed to configure missing platform");
+        return NULL;
+    }
 
-    g_autofree char *platform_name = g_strdup(name);
     g_autofree char *platform_params = g_strdup(params);
 
     if (env_prefix) {
-        if (!platform_name) {
-            g_autofree char *name_var = g_strconcat(env_prefix, "_PLATFORM_NAME", NULL);
-            platform_name = g_strdup(g_getenv(name_var));
-        }
         if (!params) {
             g_autofree char *params_var = g_strconcat(env_prefix, "_PLATFORM_PARAMS", NULL);
             platform_params = g_strdup(g_getenv(params_var));
         }
     }
-    g_debug("%s: name '%s', params '%s'", G_STRFUNC, platform_name, platform_params);
-
-    CogPlatform *platform = cog_platform_new(platform_name, error);
-    if (!platform)
-        return NULL;
```

Comment:
Moving this out of `cog_platform_configure()` leaves it with only the task of calling `cog_platform_setup()` so I am not sure it's worth it to keep the function. Maybe we can remove it... Look at my other comment, too.